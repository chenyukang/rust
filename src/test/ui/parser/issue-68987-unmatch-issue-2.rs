use std::io::Error;
fn obstest() -> Result<(), Error> {
    let obs_connect = || -> Result<(), Error) {
        //~^ ERROR mismatched closing delimiter
        //~| ERROR expected one of
        if true { Ok(()) } else { Err(Error::last_os_error()) }
    };

    Ok(())
}

fn main() {}
