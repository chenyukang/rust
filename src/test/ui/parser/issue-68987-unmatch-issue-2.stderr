error: mismatched closing delimiter: `)`
  --> $DIR/issue-68987-unmatch-issue-2.rs:1:32
   |
LL | async fn obstest() -> Result<> {
   |                                ^ unclosed delimiter
LL |     let obs_connect = || -> Result<(), MyError) {
   |                                               ^ mismatched closing delimiter

error: unexpected closing delimiter: `}`
  --> $DIR/issue-68987-unmatch-issue-2.rs:12:1
   |
LL | }
   | ^ unexpected closing delimiter

error: aborting due to 2 previous errors

